### PR TITLE
276 lane route display ticket note in history sidebar

### PR DESCRIPTION
**Pull request by HaleyBoles**
- added note to ticket history sidebar
- made sidebar virtualizers dynamic heights
- moved right sidebar to its own component

### CODE REVIEW

File: src/components/laneRoute/RightSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,1023 @@
...
+import { Popup } from "../Popup";
+import { Select } from "../Select";
+import { TicketReprint } from "../TicketReprint";
+
+const rightSidebarSelectedAtom = atom<"History" | "Pending" | "In-Yard" | "Requests">("History");
+
+// Setting up default to and from dates
+const todaysDate = new Date();
+// Can use daysBack in the future for a default range
+const daysBack = 5;
```
Comment (sspoon865) [inline src/components/laneRoute/RightSidebar.tsx:39]:
```suggestion
const daysBack = 0;
```

File: src/components/laneRoute/RightSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,1023 @@
...
+											product: ticket.orderProduct.orderProductKey,
+											poNumber: ticket.ticketPONumber ?? undefined,
+											cashReference: ticket.cashReference ?? undefined,
+											haulZone: ticket.haulZoneKey ?? undefined,
+											taxCode: ticket.taxCodeKey ?? undefined,
+											note: ticket.note ?? "",
+											udf1: ticket.udf1 ?? "",
+											udf2: ticket.udf2 ?? "",
+											udf3: ticket.udf3 ?? "",
+											activeTab: paramValues.note ? "ticketNote" : "orderInfo",
```
Comment (sspoon865) [inline src/components/laneRoute/RightSidebar.tsx:285]: Active tab for ticket notes is not getting set correctly for me. First time I click a ticket in the history sidebar with a note in it, it sets the note in the URL params, but still sets the `activeTab` to orderInfo.

Clicking the ticket a second time switching the `activeTab` to ticketNote as expected.

**Review by sspoon865 — CHANGES_REQUESTED** [review]
See comments

File: src/components/laneRoute/RightSidebar.tsx (hunk 1 above)
Comment (HaleyBoles) [reply src/components/laneRoute/RightSidebar.tsx:39]: Not again 😂

**Review by sspoon865 — APPROVED** [review]

**Pull request merged**